lib/ansible/modules/net_tools/basics/slurp.py

```diff
@@ -70,21 +90,32 @@ def main():
     module = AnsibleModule(
         argument_spec=dict(
             src=dict(type='path', required=True, aliases=['path']),
+            chunk=dict(type='int'),
         ),
         supports_check_mode=True,
     )
     source = module.params['src']
+    chunk = module.params['chunk']
 
     if not os.path.exists(source):
         module.fail_json(msg="file not found: %s" % source)
     if not os.access(source, os.R_OK):
         module.fail_json(msg="file is not readable: %s" % source)
 
     with open(source, 'rb') as source_fh:
-        source_content = source_fh.read()
-    data = base64.b64encode(source_content)
-
-    module.exit_json(content=data, source=source, encoding='base64')
+        if chunk is None or chunk <= 0:
+            data = base64.b64encode(source_fh.read())
+        else:
+            encoded = []
+            while True:
+                e_chunk = source_fh.read(chunk)
+                if not e_chunk:
+                    break
+                encoded.append(base64.b64encode(e_chunk))
+
+            data = ''.join(encoded)
```

Comment:
I think this will fail on py3, as `encoded` will be a list of bytes.

```
TypeError: sequence item 0: expected str instance, bytes found
```

Might make `b''.join` and then use `to_native(data, errors='surrogate_or_strict')`